sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -157,14 +157,34 @@ a!
 ----
 class device {
 
+bool ext_oneapi_can_build(ext::oneapi::experimental::source_language lang);
+
+};
+----
+!====
+
+_Returns:_ The value `true` only if the device supports the
+`ext::oneapi::experimental::build` function on kernel bundles written in the
+source language `lang`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+class device {
+
 bool ext_oneapi_can_compile(ext::oneapi::experimental::source_language lang);
 
 };
 ----
 !====
 
-_Returns:_ The value `true` only if the device supports kernel bundles written
-in the source language `lang`.
+_Returns:_ The value `true` only if the device supports the
+`ext::oneapi::experimental::compile` function on kernel bundles written in the
+source language `lang`.
```

Comment:
This new wording seems right.  However, it makes this existing statement in `create_kernel_bundle_from_source` unclear:

> An `exception` with the `errc::invalid` error code if the source language `lang` is not supported by any device contained by the context `ctxt`.

Options I see are:

* Change this statement to say that an exception is thrown if neither compilation nor building is supported (i.e. both `ext_oneapi_can_compile` and `ext_oneapi_can_build` return false); or
* Remove the exception entirely.  In this case we allow creating a kernel bundle for any source language and only diagnose an exception when `compile` or `build` is called.

Reply:
I'd say the second option is the better of the two, as we avoid a cases where the user can't actually compile, but tries to do so instead and doesn't get a proper failure from it. I've moved the requirement from `create_kernel_bundle_from_source` and moved it to `build` and `compile`. Interestingly, it seems like this is already the behavior of the implementation too.

Reply:
Seems reasonable.  You should also update this non-normative note in `create_kernel_bundle_from_source` to mention the `compile` function in addition to `build`:

> This function succeeds even if some devices in `ctxt` do not support the source language `lang`. However, the `build` function fails unless all of its devices support `lang`. Therefore, applications should take care to omit devices that do not support `lang` when calling `build`.